src/jdk.compiler/share/classes/com/sun/tools/javac/code/Symtab.java

```diff
@@ -475,8 +472,8 @@ public <R, P> R accept(ElementVisitor<R, P> v, P p) {
         errType = new ErrorType(errSymbol, Type.noType);
 
         unknownSymbol = new ClassSymbol(PUBLIC|STATIC|ACYCLIC, names.fromString("<any?>"), null, rootPackage);
-        unknownSymbol.members_field = new Scope.ErrorScope(unknownSymbol);
-        unknownSymbol.type = unknownType;
+        // Create the unknown type
```

Comment:
More general question: is the distinction between unknown and erroneous even relevant? It seems that an unknown type is just an erroneous type whose "class" happens to be the unknown symbol class. 

Can we just say:

```
        unknownSymbol = new ClassSymbol(PUBLIC|STATIC|ACYCLIC, names.fromString("<any?>"), null, rootPackage);
        unknownType = unknownSymbol.type = new ErrorType(unknownSymbol);
```

Is the only difference what the public `getTypeKind` will report? Do we care if there's a difference there?

Reply:
Internally, after this patch, the distinction is minimal, and usually checked using `== unknownType`. Whether we can change the external behavior not sure. We can try.